examples/diesel-mysql/main.rs

```diff
@@ -18,72 +18,55 @@ fn main() -> anyhow::Result<()> {
         "mysql://demo:password@localhost/jiff_diesel_example",
     )?;
 
-    example_datetime_decode(&mut conn)?;
-    example_datetime_encode(&mut conn)?;
+    example_datetime_roundtrip(&mut conn)?;
 
     Ok(())
 }
 
-/// Performs a decoding with all of Jiff's datetime types.
-fn example_datetime_decode(conn: &mut MysqlConnection) -> anyhow::Result<()> {
-    // Decoding jiff::Timestamp
-    let query = select(sql::<sql_types::Datetime>("'1970-01-01 00:00:00Z'"));
-    let got: jiff_diesel::Timestamp = query.get_result(conn)?;
-    let expected =
-        "1970-01-01T00:00:00Z".parse::<jiff::Timestamp>()?.to_diesel();
-    assert_eq!(expected, got);
+/// Performs a round-trip with all of Jiff's datetime types.
+fn example_datetime_roundtrip(
+    conn: &mut MysqlConnection,
+) -> anyhow::Result<()> {
+    diesel::table! {
+        jiffs {
+            id -> Integer, // Diesel tables require an ID column.
+            ts -> Datetime,
+            dt -> Timestamp,
+            d -> Date,
+            t -> Time
+        }
+    }
 
-    // Decoding jiff::civil::DateTime
-    let query = select(sql::<sql_types::Timestamp>("'2025-07-20 00:00:00'"));
-    let got: jiff_diesel::DateTime = query.get_result(conn)?;
-    let expected = civil::date(2025, 7, 20).at(0, 0, 0, 0).to_diesel();
-    assert_eq!(expected, got);
+    #[derive(Debug, PartialEq, QueryableByName)]
+    #[diesel(check_for_backend(diesel::mysql::Mysql))]
+    struct Jiff {
+        #[diesel(deserialize_as = jiff_diesel::Timestamp)]
+        pub ts: jiff::Timestamp,
+        #[diesel(deserialize_as = jiff_diesel::DateTime)]
+        pub dt: jiff::civil::DateTime,
+        #[diesel(deserialize_as = jiff_diesel::Date)]
+        pub d: jiff::civil::Date,
+        #[diesel(deserialize_as = jiff_diesel::Time)]
+        pub t: jiff::civil::Time,
```

Comment:
Do these fields need to be `pub`? If not, I'd remove it.

Reply:
Done.